clients/src/main/java/org/apache/kafka/common/errors/ObsoleteBrokerException.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership. The ASF licenses this file
+ * to You under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package org.apache.kafka.common.errors;
+
+/**
+ * Indicates that a request cannot be completed because an obsolete broker
```

Comment:
Hmm.  The problem with UnsupportedBrokerVersion is that it doesn't make it clear whether the broker version is too new or old.  It also suggests that the whole broker is unsupported, whereas just this one API is unsupported.

> Another point: if we think of the protocol as something independent from Kafka (i.e. there could be other implementations), which is something that Jay thinks we should, then UnsupportedBrokerVersion also seems better.

Hmm.  It seems like we could still have other implementations of the protocol even if we used OutdatedBrokerException / ObsoleteBrokerException?